src/cacao_zcap/mod.rs

```diff
@@ -0,0 +1,206 @@
+                Err(err) => {
+                    return (ResolutionMetadata::from_error(&err.to_string()), None, None);
+                }
+            };
+            (
+                Default::default(),
+                Some(doc),
+                Some(DocumentMetadata::default()),
+            )
+        }
+    }
+
+    #[async_std::test]
+    async fn siwe_verify() {
+        let message = Message::from_str(
+            r#"localhost:4361 wants you to sign in with your Ethereum account:
+0x6Da01670d8fc844e736095918bbE11fE8D564163
+
+SIWE Notepad Example
+
+URI: http://localhost:4361
+Version: 1
+Chain ID: 1
+Nonce: kEWepMt9knR6lWJ6A
+Issued At: 2021-12-07T18:28:18.807Z"#,
+        )
+        .unwrap();
+        let payload = Payload::from(message);
+        // Sanity check: verify signature
+        let sig_mb = r#"f6228b3ecd7bf2df018183aeab6b6f1db1e9f4e3cbe24560404112e25363540eb679934908143224d746bbb5e1aa65ab435684081f4dbb74a0fec57f98f40f5051c"#;
+        let (_base, sig) = multibase::decode(&sig_mb).unwrap();
+        let sig = BasicSignature {
+            s: sig.try_into().unwrap(),
+        };
+        let cacao = CACAO::<SignInWithEthereum>::new(payload, sig);
+        cacao.verify().await.unwrap();
+        // This SIWE is not expected to be valid as a CACAO Zcap, but is here as an example that is
+        // verifiable.
+    }
+
+    #[async_std::test]
+    async fn zcap_cacao_kepler_session() {
+        let siwe_msg_str = include_str!("../../tests/cacao_zcap/delegation0.siwe");
+        let siwe_msg_sig_hex = include_str!("../../tests/cacao_zcap/delegation0.siwe.sig");
+        let siwe_msg = Message::from_str(siwe_msg_str).unwrap();
+        let payload = Payload::from(siwe_msg);
+        let (_base, sig) = multibase::decode(&format!("f{}", siwe_msg_sig_hex)).unwrap();
+        let sig = BasicSignature {
+            s: sig.try_into().unwrap(),
+        };
+        let cacao = CACAO::<SignInWithEthereum>::new(payload, sig);
+        let zcap = cacao_to_zcap(&cacao).unwrap();
+        let zcap_json = serde_json::to_value(&zcap).unwrap();
+        let zcap_json_expected: Value = serde_json::from_str(include_str!(
+            "../../tests/cacao_zcap/delegation0-zcap.jsonld"
+        ))
+        .unwrap();
+        assert_eq!(zcap_json, zcap_json_expected);
+
+        let _resolver = ExampleDIDPKH;
+        // Verify cacao as zcap
+        /* Can't call zcap.verify yet because that depends on ssi
+         * having this proof type.
+        use crate::vc::Check;
+        let res = zcap.verify(None, &resolver).await;
+        assert_eq!(res.errors, Vec::<String>::new());
+        assert!(res.checks.iter().any(|c| c == &Check::Proof));
+        */
+
+        /* Can't verify because signature is not real
+        let proof = zcap.proof.as_ref().unwrap();
+        let warnings = CacaoZcapProof2022
+            .verify(proof, &zcap, &resolver)
+            .await
+            .unwrap();
+        dbg!(warnings);
+        */
```

Comment:
Can this be uncommented now that it has been added as a proof type?

Reply:
Updated the comment in e1cbf6c32a48c0448cb8c4c3f7d8bfdcf80f0548
The second part "Can't verify because signature is not real" is still true; a real signed test vector will need to be created to actually test verify. `examples/cacao-zcap-regenerate-test-vectors.rs` could be updated to do that.